coreutils_core/src/settime.rs

```diff
@@ -0,0 +1,21 @@
+use libc::timeval;
+use std::ptr;
+use errno::errno;
+
+/// Wrapper function for `libc::settimeofday`
+pub fn settimeofday(tv_sec: i64, tv_nsec: i32) -> Result<(), String> {
+    let timeval = timeval {
+        tv_sec,
+        tv_usec: tv_nsec
+    };
+
+    unsafe {
+        let result = libc::settimeofday(&timeval as *const timeval, ptr::null());
+        match result {
+            0 => Ok(()),
+            _ => {
+                Err(errno().to_string())
+            }
+        }
+    }
+}
```

Comment:
```rust
let result = unsafe { libc::settimeofday(&timeval as *const timeval, ptr::null()) };
```
And the match outside of the block

Also, it would be better to return `io::Result` and get the last os error, since they are pretty much all possible system error, would be equivalent to use errno, but more "rustic"

Reply:
Okay fixed this. But what do you mean with the `tv_nsec` and `tv_usec`?

Reply:
I'm not 100% sure, but in timeval `tv_usec` is microseconds while `tv_nsec` should mean nanoseconds, so it raises confusion on my mind.

In *nix we usually have 
```rust
struct timeval {
    tv_sec: <integer>,
    tv_usec: <integer>,
}

struct timestamp {
    tv_sec: <integer>,
    tv_nsec: <integer>,
}
```
So it was confusing what you actually wanted, plus, looks wrong putting something that is supposed to be `nsec` on a field that wants `usec`